### PR TITLE
OSD: Allow loading of fonts from filesystem

### PR DESCRIPTION
**Pull request by tridge**
This allows users to load fonts from their microSD. They can either replace an existing font number or add a new font number from 0 to 9.
For example, here I have put the chinese font from #15668 as font5.bin on the microSD, and set OSD_FONT to 5.
This also adds a general load_file() API to AP_Filesystem which makes loading small files into memory easy.

image

This will also allow us to free up flash by only including one font

### CODE REVIEW

**Comment by vierfuffzig** [comment]
awesome, thanks @tridge

File: libraries/AP_Filesystem/AP_Filesystem_backend.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{
+    struct stat st;
+    if (stat(filename, &st) != 0) {
+        return nullptr;
+    }
+    FileData *fd = new FileData(this);
+    if (fd == nullptr) {
+        return nullptr;
+    }
+    void *data = (uint8_t *)malloc(st.st_size);
```
Comment (peterbarker) [inline libraries/AP_Filesystem/AP_Filesystem_backend.cpp:31]: Pointless cast?

File: libraries/AP_Filesystem/AP_Filesystem_backend.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    return fd;
+}
+
+/*
+  unload a FileData object
+*/
+void AP_Filesystem_Backend::unload_file(FileData *fd)
+{
+    if (fd->data != nullptr) {
+        free(const_cast<uint8_t *>(fd->data));
```
Comment (peterbarker) [inline libraries/AP_Filesystem/AP_Filesystem_backend.cpp:60]: Free takes a `void*`....

Comment (peterbarker) [reply libraries/AP_Filesystem/AP_Filesystem_backend.cpp:60]: Should probably assign nullptr to fd->data

File: libraries/AP_Filesystem/AP_Filesystem_ROMFS.cpp [hunk 3]
```diff
@@ -212,4 +212,25 @@ bool AP_Filesystem_ROMFS::set_mtime(const char *filename, const uint32_t mtime_s
...
 }
 
+/*
+  load a full file. Use delete to free the data
+  we override this in ROMFS to avoid taking twice the memory
+*/
+FileData *AP_Filesystem_ROMFS::load_file(const char *filename)
+{
+    FileData *fd = new FileData(this);
+    fd->data = AP_ROMFS::find_decompress(filename, fd->length);
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_Filesystem/AP_Filesystem_ROMFS.cpp:222]: fixed

File: libraries/AP_Filesystem/AP_Filesystem_backend.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Filesystem/AP_Filesystem_backend.cpp:31]: fixed

File: libraries/AP_Filesystem/AP_Filesystem_backend.cpp (hunk 2 above)
Comment (tridge) [reply libraries/AP_Filesystem/AP_Filesystem_backend.cpp:60]: can't be void*, not allowed by const_cast from a uint8_t*

**Pull request merged**